docs/Testing.md

```diff
@@ -10,12 +10,156 @@ This isn't a hard and fast rule, but it does express our approach fairly well.
+
+Then, inside the test function, we can make "assertions", or what we expect a value to be using the `expect` function. We pass the value as the first argument to the `expect` function, then we use one of the "matcher" methods on `expect`, such as `.toBe` to describe what the value should match.
+
+[The React Native doc](https://reactnative.dev/docs/testing-overview#unit-tests) provide the following example for a unit test:
+
+```ts
+it("given a date in the past, colorForDueDate() returns red", () => {
+  const input = colorForDueDate("2000-10-20")
+  expect(input).toBe("red")
+})
+```
+
+Jest functions like `it`, `test`, `expect`, and more are loaded globally by the Jest test runner, so you don't need to import them.
+
+### Best Practices
+
+> When writing a test, do your best to make sure that your tests > include the following information:
+>
+> - Given - some precondition
+> - When - some action executed by the function that you’re testing
+> - Then - the expected outcome
+>   This is also known as AAA (Arrange, Act, Assert).
+
+-- <cite>[Structuring Tests in React Native Docs](https://reactnative.dev/docs/testing-overview#structuring-tests)</cite>
+
+You can read more about how to best practices for creating tests in the [Structuring Tests](https://reactnative.dev/docs/testing-overview#structuring-tests) section of the React Native docs.
+
+### Writing Tests
+
+To write your own tests, create a `.test.ts` file within the `app` or `test` directory.
+
+Then run `yarn test` to run all unit tests using Jest.
+
+When writing tests, you can also run Jest in watch mode by running `yarn test:watch`. This will start a long running Jest process, that re-runs your tests on save in your editor. This is useful for iterating on values and getting quick feedback about whether your changes were successful or not.
+
+### When to write Unit Tests
+
+Some developers believe that every line of code should have a co-responding test. However, Infinite Red believes that some unit tests provide more values than others. Here are a few situations where we have found unit tests to be helpful in client projects:
+
+- **complicated regex's**: for many developers, regex's only make sense as you are writing them. Testing with a series of valid and invalid inputs can help ensure that they work as intended for future developers.
+- **nested if/else statements**: if you rely heavily on a function with a lot of if/else statements, it can be helpful to have tests to make sure that you can visit each condition. Often times when we have more than a handful of conditionals, it can become impossible to visit all of them without realizing it.
+- **validation functions**: often times we may write functions like `isJson()` to validate that a value is a specific shape. If critical parts of our code rely on the correctness of this function, it can be useful to add tests to make sure that it fails on invalid inputs and passes on valid inputs.
```

Comment:
Think this can be more concise, one idea:
```suggestion
- **validation functions**: often times we may write functions like `isJson()` to validate that a value is a specific shape. If critical parts of our code rely on the correctness of this function, we want to test that code!
```